### DISCUSSION TITLE
Where to find (or create) Site urls and Additional Redirect URLs - Flutter

### DISCUSSION THREAD

**Discussion by devberkay** [supabase/supabase#5947] [Questions]
Hey, i am newbie on Supabase and following the official documents for authentication procedures for flutter and i couldn't understand how to find (or create) site urls and additional redirect urls for my mobile app : 

![image](https://user-images.githubusercontent.com/81699676/158809768-2cbf0de4-168e-4c64-a226-049220203f97.png)

here is the official flutter example on authentication via deeplinks and it edits **additional redirect urls** but doesn't explain where to find : [https://supabase.com/docs/guides/with-flutter](url)

here is another official document that explains how to  authenticate via email & password and it edits **site urls** 
but doesn't explain where to create & find : https://supabase.com/docs/guides/auth/auth-email

**Comment by GaryAustin1** [comment]
The site url is your apps url on the web  `https://mygreatapp.com` or if you are doing development locally `http://local` host:3000`.    The redirect url is the full url of a sub path in your code you want confirmation email to redirect user to.  These are part of your code base structure.

**Reply by devberkay** [reply]
For example, i want my app's users to be redirected back to my app after they clicked on the confirmation e-mail, for such a situation, will i submit my application's store link there ? ( currently i don't have it because i am still on development stage and not submitted the app to the google or apple)

**Reply by SunilKividor** [reply]
@devberkay did you find any solution for this ?

**Comment by GaryAustin1** [comment]
The url for native apps is more complicated, but basically it is the url to get to the correct "page" in your app on THE USER's DEVICE.
Here is an example of flutter and deep linking: https://github.com/supabase-community/supabase-flutter/blob/main/README.md#deeplink-config

**Comment by aleksey-mukho** [comment]
@devberkay You can set up universal links for web and mobile

**Comment by dhurbhat** [comment]
Like @devberkay, I too am quite confused on deep-linking with password or any other email confirmation works for **mobile apps**.

Say my site url is `http://localhost:3000` or `https://io.supabase.flutterquickstart/` - then with the Confirm email, the site_url appears to be used to accept the confirmation, meaning that email message brings the user to the site URL along with access token data.  This makes sense...

For deep linking, I've tried:
* updating AndroidManifest.xml with the intent as described https://supabase.com/docs/guides/getting-started/tutorials/with-flutter#setup-deep-links
* updated `info.plist` for iOS again as described in above link.
* also added `authCallbackUrlHostname: 'io.supabase.flutterquickstart'` to `supabase.initialize()` - BTW I'm using Flutter

BUT it doesn't redirect to the deep-link... it always goes to the web (which works when I set `site url` to `localhost` but fails with `io.supabase.flutterquickstart`...

What am I missing?  What's the magic formula here.  

Truly wish there were succinct documentation **OR BETTER YET A SEQUENCE DIAGRAM**

With the Supabase Flutter Auth UI app installed locally, AND running in Android, I would like:
1. User signs up on mobile (Android or iOS)
2. User receives confirmation message on the mobile device... then clicking on the confirmation link DEEP LINKS to the specified page `/home` on the app

What configurations for `site url` and `redirect url` should I use presuming I **DON'T CARE FOR WEB** ???

**Reply by paulmbw** [reply]
Is this thread still alive? I'm also not sure what `Site URL` should be when my app is deployed onto prod (and uploaded to the app store)

**Reply by chrisdemetriad** [reply]
Me neither, it's so frustrating!

**Reply by Ishpurwar-dev** [reply]
Is this thread still alive? I am not able to find `Site URL`  and `redirect URL` of my flutter app that is running on my mobile emulator. I am able to receive email confirmation and currently i have set my Site URL to this `io.supabase.flutterquickstart` and on click confirmation URL it take me to web page which show invalid page.
